src/platforms/common/cdcacm.c

```diff
@@ -419,7 +419,8 @@ static const char *usb_strings[] = {
 	serial_no,
 	"Black Magic GDB Server",
 	"Black Magic UART Port",
-	"Black Magic DFU",
+	/* Required, the line for stm32f1s is "@Internal Flash   /0x08000000/8*001Ka,56*001Kg" maybe use an if statement? */
+	"@Internal Flash   /0x08000000/04*016Kg,01*064Kg,03*128Kg",
```

Comment:
Having further reviewed all this code, we can with confidence say that this patch is not required and that if anything was, it would go in src/platforms/common/stm32/dfucore.c rather than here as the strings apply to DFU mode not application mode and so apply to the bootloader and not the main firmware:

https://github.com/blackmagic-debug/blackmagic/blob/8e661a6887fd8d8dba459e08264d25406067656e/src/platforms/common/stm32/dfucore.c#L36-L45

These definitions do look like they need a review though so you would be most welcome to figure out what's going on there and suggest any required patches in this PR

Reply:
ha my god my memory is bad it was so long ago. The idea which I probly never said so my bad, was to use the internal dfu bootloader on models that support it. Using the stm dfu this would be required for the switching back and forth to work correctly. The usb descriptors for dfu would need to be set this way in the main firmware so upon reset into dfu mode dfu-util would know it was same dfu device. My idea was the in house dfu firmware wasn't needed, and using the internal dfu would save space. But using the inhouse dfu became the decided direction (can't blame yeah, tho I am curious for the reason, just so I can use that knowledge to become a better programmer/contributer)